From df3db28a0dd7ee452a5992493a28a345f4096343 Mon Sep 17 00:00:00 2001
From: Rohit Athavale <rohit.athavale@xilinx.com>
Date: Tue, 20 Feb 2018 10:14:08 -0800
Subject: [PATCH 0284/1852] media: xilinx-vpss-scaler: Zero out H-phase array
 before changing resolution

commit bbf6343857aa2bfdc5f63d7d7218086ceb61002f from
https://github.com/Xilinx/linux-xlnx.git

When changing resolutions, the H-phases need to be re-calculated. This
commit zeroes out the H-phase array. This fixes the stretching
(pixel repeat) seen when upscaling video.

Signed-off-by: Rohit Athavale <rohit.athavale@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-scaler.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
index cfcefe6d673a..5fc41b940643 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
@@ -1325,6 +1325,7 @@ static int xscaler_s_stream(struct v4l2_subdev *subdev, int enable)
 		gpiod_set_value_cansleep(xscaler->rst_gpio,
 					 XSCALER_RESET_DEASSERT);
 		xscaler_reset(xscaler);
+		memset(xscaler->H_phases, 0, sizeof(xscaler->H_phases));
 		return 0;
 	}
 
-- 
2.31.1

